osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs

```diff
@@ -0,0 +1,212 @@
+                    },
+                    referenceScoreContainer = new FillFlowContainer
+                    {
+                        Spacing = new Vector2(10),
+                        Direction = FillDirection.Vertical,
+                        RelativeSizeAxes = Axes.X,
+                        AutoSizeAxes = Axes.Y,
+                    },
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            ReferenceScore.BindValueChanged(scoreChanged, true);
+
+            beatmapOffsetSubscription = realm.RegisterCustomSubscription(r =>
+            {
+                var userSettings = r.Find<BeatmapInfo>(beatmap.Value.BeatmapInfo.ID)?.UserSettings;
+
+                if (userSettings == null) // only the case for tests.
+                    return null;
+
+                Current.Value = userSettings.Offset;
+                userSettings.PropertyChanged += onUserSettingsOnPropertyChanged;
+
+                return new InvokeOnDisposal(() => userSettings.PropertyChanged -= onUserSettingsOnPropertyChanged);
+
+                void onUserSettingsOnPropertyChanged(object sender, PropertyChangedEventArgs args)
+                {
+                    if (args.PropertyName == nameof(BeatmapUserSettings.Offset))
+                        Current.Value = userSettings.Offset;
+                }
+            });
+
+            Current.BindValueChanged(currentChanged);
+        }
+
+        private Task realmWrite;
+
+        private void currentChanged(ValueChangedEvent<double> offset)
+        {
+            Scheduler.AddOnce(updateOffset);
+
+            void updateOffset()
+            {
+                // ensure the previous write has completed. ignoring performance concerns, if we don't do this, the async writes could be out of sequence.
+                if (realmWrite?.IsCompleted == false)
+                {
+                    Scheduler.AddOnce(updateOffset);
+                    return;
+                }
+
+                if (useAverageButton != null)
+                    useAverageButton.Enabled.Value = !Precision.AlmostEquals(lastPlayAverage, Current.Value, Current.Precision);
+
+                realmWrite = realm.WriteAsync(r =>
+                {
+                    var settings = r.Find<BeatmapInfo>(beatmap.Value.BeatmapInfo.ID)?.UserSettings;
+
+                    if (settings == null) // only the case for tests.
+                        return;
+
+                    if (Precision.AlmostEquals(settings.Offset, Current.Value))
+                        return;
+
+                    settings.Offset = Current.Value;
+                });
+            }
+        }
+
+        private void scoreChanged(ValueChangedEvent<ScoreInfo> score)
+        {
+            referenceScoreContainer.Clear();
+
+            if (score.NewValue == null)
+                return;
+
+            if (score.NewValue.Mods.Any(m => m is ModAutoplay))
```

Comment:
I think the more correct thing to check here would be

```suggestion
            if (score.NewValue.Mods.Any(m => !m.IsUserPlayable))
```

Reply:
That's the one I was looking for.